src/control/server/security_rpc_test.go

```diff
@@ -0,0 +1,363 @@
+		Status: int32(drpc.DaosInvalidInput),
+	})
+}
+
+func TestSrvSecurityModule_ValidateCred_NoToken(t *testing.T) {
+	log, buf := logging.NewTestLogger(t.Name())
+	defer common.ShowBufferOnFailure(t, buf)
+
+	mod := NewSecurityModule(log, insecureTransportConfig())
+	reqBytes := getMarshaledValidateCredReq(t, nil, &auth.Token{
+		Flavor: auth.Flavor_AUTH_NONE,
+		Data:   []byte{byte(1), byte(2)},
+	})
+
+	resp, err := callValidateCreds(mod, reqBytes)
+
+	if err != nil {
+		t.Errorf("Expected no error, got: %v", err)
+	}
+
+	expectValidateResp(t, resp, &auth.ValidateCredResp{
+		Status: int32(drpc.DaosInvalidInput),
+	})
+}
+
+func TestSrvSecurityModule_ValidateCred_NoVerifier(t *testing.T) {
+	log, buf := logging.NewTestLogger(t.Name())
+	defer common.ShowBufferOnFailure(t, buf)
+
+	mod := NewSecurityModule(log, insecureTransportConfig())
+	reqBytes := getMarshaledValidateCredReq(t, &auth.Token{
+		Flavor: auth.Flavor_AUTH_NONE,
+		Data:   []byte{byte(1), byte(2)},
+	}, nil)
+
+	resp, err := callValidateCreds(mod, reqBytes)
+
+	if err != nil {
+		t.Errorf("Expected no error, got: %v", err)
+	}
+
+	expectValidateResp(t, resp, &auth.ValidateCredResp{
+		Status: int32(drpc.DaosInvalidInput),
+	})
+}
+
+func getValidToken(t *testing.T) *auth.Token {
+	tokenData := &auth.Sys{
+		Stamp: uint64(time.Now().Unix()),
+		User:  "gooduser@",
+		Group: "goodgroup@",
+	}
+	return &auth.Token{
+		Flavor: auth.Flavor_AUTH_SYS,
+		Data:   marshal(t, tokenData),
+	}
+}
+
+func getVerifierForToken(t *testing.T, token *auth.Token, key crypto.PublicKey) *auth.Token {
+	verifier, err := auth.VerifierFromToken(key, token)
+	if err != nil {
+		t.Fatalf("Couldn't get verifier: %v", err)
+	}
+
+	return &auth.Token{
+		Flavor: auth.Flavor_AUTH_SYS,
+		Data:   verifier,
+	}
+}
+
+func TestSrvSecurityModule_ValidateCred_Insecure_OK(t *testing.T) {
+	log, buf := logging.NewTestLogger(t.Name())
+	defer common.ShowBufferOnFailure(t, buf)
+
+	mod := NewSecurityModule(log, insecureTransportConfig())
+
+	token := getValidToken(t)
+	reqBytes := getMarshaledValidateCredReq(t, token, getVerifierForToken(t, token, nil))
+
+	resp, err := callValidateCreds(mod, reqBytes)
+
+	if err != nil {
+		t.Errorf("Expected no error, got: %v", err)
+	}
+
+	expectValidateResp(t, resp, &auth.ValidateCredResp{
+		Token: token,
+	})
+}
+
+func TestSrvSecurityModule_ValidateCred_Insecure_BadVerifier(t *testing.T) {
+	log, buf := logging.NewTestLogger(t.Name())
+	defer common.ShowBufferOnFailure(t, buf)
+
+	mod := NewSecurityModule(log, insecureTransportConfig())
+
+	token := getValidToken(t)
+	reqBytes := getMarshaledValidateCredReq(t, token, &auth.Token{Data: []byte{0x1}}) // junk verifier
+
+	resp, err := callValidateCreds(mod, reqBytes)
+
+	if err != nil {
+		t.Errorf("Expected no error, got: %v", err)
+	}
+
+	expectValidateResp(t, resp, &auth.ValidateCredResp{
+		Status: int32(drpc.DaosNoPermission),
+	})
+}
+
+func generateTestCert(t *testing.T, dir string) crypto.PrivateKey {
```

Comment:
This will work for now but we will have to add certificate validation in the cert loading code at some point for the data plane and this will probably need to be signed by a CA Cert. But for now it is a good start.

Reply:
It looks like this cert contains both a public and private key is that the case? Or is it creating two files?

Reply:
It's writing the cert file, but doesn't write the private key to a file. My test only needed to read the cert from the file system, so I just did the minimum required. The private key is returned to the test (which needs to sign with it to create a valid credential).

Reply:
Looks good.